### PR TITLE
Debug error: Identifier now has optional `referenced_declaration` field

### PR DESCRIPTION
**Pull request by TilakMaddy**
@alexroan Please try it out on this branch and let me know 🙏 

It should work

### CODE REVIEW

File: aderyn_core/src/ast/identifiers.rs [hunk 1]
```diff
@@ -11,7 +11,7 @@ pub struct Identifier {
     pub argument_types: Option<Vec<TypeDescriptions>>,
     pub name: String,
     pub overloaded_declarations: Vec<NodeID>,
-    pub referenced_declaration: NodeID,
+    pub referenced_declaration: Option<NodeID>,
     pub type_descriptions: TypeDescriptions,
```
Comment (TilakMaddy) [inline aderyn_core/src/ast/identifiers.rs:15]: main change here

**Comment by TilakMaddy** [comment]
@alexroan This works on the lssvm codebase 

Follow up for me -
This Fix should be done "properly" in a different PR like "Part 1.5". That will be the new pre-requisite for Part 2 & Part 3

**Comment by alexroan** [comment]
Sweet. Ping me when it's ready @TilakMaddy :) I'll get to it ASAP.

**Comment by TilakMaddy** [comment]
Closing in favor #549

**Pull request closed without merging**